go/chat/commands/location.go

```diff
@@ -0,0 +1,74 @@
+package commands
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+)
+
+type Location struct {
+	*baseCommand
+	sync.Mutex
+	displayed bool
+}
+
+func NewLocation(g *globals.Context) *Location {
+	return &Location{
+		baseCommand: newBaseCommand(g, "location", "", "Post your current location", true),
+	}
+}
+
+func (n nullChatUI) ChatGetCoordinate(ctx context.Context) (chat1.Coordinate, error) {
+	return chat1.Coordinate{}, errors.New("no UI available")
+}
+
+func (h *Location) Execute(ctx context.Context, uid gregor1.UID, convID chat1.ConversationID,
+	tlfName, text string, replyTo *chat1.MessageID) (err error) {
+	defer h.Trace(ctx, func() error { return err }, "Location")()
+	if !h.Match(ctx, text) {
+		return ErrInvalidCommand
+	}
+	coord, err := h.getChatUI().ChatGetCoordinate(ctx)
+	if err != nil {
+		return err
+	}
+	if _, err := h.G().ChatHelper.SendMsgByIDNonblock(ctx, convID, tlfName,
+		chat1.NewMessageBodyWithText(chat1.MessageText{
+			Body:  "_Sharing my location..._",
```

Comment:
is this hard to remove because you can't send a text with an empty body? could we modify to allow if `Coord != nil`?

Reply:
Discussed offline